### ISSUE TITLE
closure_spacing should ignore empty bodies

### ISSUE THREAD

**Issue by bensarz** [realm/SwiftLint#875]
Triggering example:

```swift

class CustomResponse: Response {
        override init() { } // triggers a warning
}
```

The expectation would be that empty bodies with a single space does not trigger a warning.

**Comment by DonSwet** [comment]
I can't think of a single real-world use-case where this would be the case. Please correct me if I'm missing something obvious. Your current example doesn't follow convention as typically you'd write:
`class CustomResponse: Response {
    override init() {
        super.init()
    }
}`

**Comment by bensarz** [comment]
Well it's not so much about the init, but about the spacing between the brackets. However, when mocking classes for testing, it often happens that you have to implement a method without a body. Using `{ }` (single space) triggers a warning and you have to use two spaces to suppress it like `{  }`.

It's a super big deal, but figured I'd open an issue anyway so it can be reviewed.

**Comment by marcelofabri** [comment]
Another common case is to make existing types conform to a protocol, for example:

```swift
protocol ReusableView: class {
    static var reuseIdentifier: String { get }
}

extension ReusableView where Self: UIView {
    static var reuseIdentifier: String {
        return String(describing: self)
    }
}

extension UITableViewCell: ReusableView { }
```
